agent-connection/src/main/kotlin/com/luxoft/blockchainlab/corda/hyperledger/indy/AgentConnection.kt

```diff
@@ -1,39 +1,47 @@
 package com.luxoft.blockchainlab.corda.hyperledger.indy
 
 import com.luxoft.blockchainlab.hyperledger.indy.models.*
+import rx.Single
+import rx.Subscription
 
 
 enum class AgentConnectionStatus { AGENT_CONNECTED, AGENT_DISCONNECTED }
 
-data class IndyParty(val did: String, val endpoint: String, val verkey: String? = null)
+interface IndyPartyConnection {
+    fun sendCredentialOffer(offer: CredentialOffer)
+    fun receiveCredentialOffer(): Single<CredentialOffer>
+
+    fun sendCredentialRequest(request: CredentialRequestInfo)
+    fun receiveCredentialRequest(): Single<CredentialRequestInfo>
+
+    fun sendCredential(credential: CredentialInfo)
+    fun receiveCredential(): Single<CredentialInfo>
+
+    fun sendProofRequest(request: ProofRequest)
+    fun receiveProofRequest(): Single<ProofRequest>
+
+    fun sendProof(proof: ProofInfo)
+    fun receiveProof(): Single<ProofInfo>
+
+    fun partyDID(): String
+    fun myDID(): String
+}
 
 class AgentConnectionException(obj: Any) :
         RuntimeException("AgentConnection exception: $obj")
 
+fun <T> Single<T>.handle(handle: (message: T?, ex: Throwable?) -> Unit): Subscription = this.subscribe({ handle(it, null) }, { handle(null, it) })
+
 interface AgentConnection {
```

Comment:
Could you provide description of methods in AgentConnection and IndyPartyConnection interfaces

Reply:
done.